drivers/serial/uart_cc13xx_cc26xx.c

```diff
@@ -401,69 +398,40 @@ static int postNotifyFxn(unsigned int eventType, uintptr_t eventArg,
 
 #ifdef CONFIG_PM_DEVICE
 static int uart_cc13xx_cc26xx_set_power_state(const struct device *dev,
-					      enum pm_device_state new_state)
+					      enum pm_device_state state)
 {
 	int ret = 0;
 
-	if ((new_state == PM_DEVICE_STATE_ACTIVE) &&
-		(new_state != get_dev_data(dev)->pm_state)) {
-		if (get_dev_conf(dev)->regs ==
-			DT_INST_REG_ADDR(0)) {
+	if (state == PM_DEVICE_STATE_ACTIVE) {
+		if (get_dev_conf(dev)->regs == DT_INST_REG_ADDR(0)) {
 			Power_setDependency(PowerCC26XX_PERIPH_UART0);
 		} else {
 			Power_setDependency(PowerCC26X2_PERIPH_UART1);
 		}
 		/* Configure and enable UART */
 		ret = uart_cc13xx_cc26xx_configure(dev,
 			&get_dev_data(dev)->uart_config);
-		if (ret == 0) {
-			get_dev_data(dev)->pm_state = new_state;
-		}
 	} else {
-		__ASSERT_NO_MSG(new_state == PM_DEVICE_STATE_LOW_POWER ||
-			new_state == PM_DEVICE_STATE_SUSPEND ||
-			new_state == PM_DEVICE_STATE_OFF);
-
-		if (get_dev_data(dev)->pm_state == PM_DEVICE_STATE_ACTIVE) {
-			UARTDisable(get_dev_conf(dev)->regs);
-			/*
-			 * Release power dependency - i.e. potentially power
-			 * down serial domain.
-			 */
-			if (get_dev_conf(dev)->regs ==
-			    DT_INST_REG_ADDR(0)) {
-				Power_releaseDependency(
-					PowerCC26XX_PERIPH_UART0);
-			} else {
-				Power_releaseDependency(
-					PowerCC26X2_PERIPH_UART1);
-			}
-			get_dev_data(dev)->pm_state = new_state;
+		UARTDisable(get_dev_conf(dev)->regs);
+		/*
+		 * Release power dependency - i.e. potentially power
+		 * down serial domain.
+		 */
+		if (get_dev_conf(dev)->regs ==
+			DT_INST_REG_ADDR(0)) {
+			Power_releaseDependency(PowerCC26XX_PERIPH_UART0);
+		} else {
+			Power_releaseDependency(PowerCC26X2_PERIPH_UART1);
 		}
 	}
 
 	return ret;
 }
 
 static int uart_cc13xx_cc26xx_pm_control(const struct device *dev,
-					 uint32_t ctrl_command,
-					 enum pm_device_state *state)
+					 enum pm_device_state state)
```

Comment:
there's an upcoming patch cleaning up driver implementations (https://github.com/zephyrproject-rtos/zephyr/pull/36125/commits/2aecc31fbbed9b98cb2d0af918a66b3d43c000c1), it's out of the scope of this PR.

Reply:
That is not out of this scope, since it is this patch that is causing this function no longer be needed. It is definitely out of scope, but I can live with it being addressed in another pr.